### DISCUSSION TITLE
Set env values conditionally

### DISCUSSION THREAD

**Discussion by pkumarch** [community/community#25625] [Actions]
<p>I want to set env variables based on input parameter. However, the variables are not available in steps following the definitions.</p>
<pre><code class="lang-auto">    steps:
    - uses: actions/checkout@v2
    - name: set environment variables
      run: |
        if [[ github.event.inputs.env_name == "dev" ]]; then 
          echo "ENVIRONMENT_NAME=${{ secrets.DEV_ENVIRONMENT_NAME }}" &gt;&gt; $GITHUB_ENV
          echo "KUBE_CONFIG=${{ secrets.DEV_KUBE_CONFIG }}" &gt;&gt; $GITHUB_ENV
          echo "KUBE_CONTEXT=${{ secrets.DEV_KUBE_CONTEXT }}" &gt;&gt; $GITHUB_ENV
        elif [[ github.event.inputs.env_name == "qc" ]]; then 
          echo "ENVIRONMENT_NAME=${{ secrets.QC_ENVIRONMENT_NAME }}" &gt;&gt; $GITHUB_ENV
          echo "KUBE_CONFIG=${{ secrets.QC_KUBE_CONFIG }}" &gt;&gt; $GITHUB_ENV
          echo "KUBE_CONTEXT=${{ secrets.QC_KUBE_CONTEXT }}" &gt;&gt; $GITHUB_ENV
       fi

    - name: Print ENV
      run: echo env
</code></pre>
<p>Variables set in above steps are not listed. It always fails to access any of the variables defined in the block.<br>
What am I missing ?</p>

**Comment by pkumarch** [comment, marked as answer]
<p>Resolved it with following approach:</p>
<pre><code class="lang-auto">env:
      ENVIRONMENT_NAME: ${{ secrets[format('{0}_ENVIRONMENT_NAME', github.event.inputs.env_name)] }}
      KUBE_CONFIG: ${{ secrets[format('{0}_KUBE_CONFIG', github.event.inputs.env_name)] }}
      KUBE_CONTEXT: ${{ secrets[format('{0}_KUBE_CONTEXT', github.event.inputs.env_name)] }}
    
</code></pre>
<p>Thanks to examples in <a href="https://sanderknape.com/2021/01/go-crazy-github-actions/" class="inline-onebox" rel="noopener nofollow ugc">Go crazy with GitHub Actions - Sander Knape</a></p>

**Comment by anomiex** [comment]
<p>Your problem in the original attempt is in this code:</p>
<pre><code class="lang-auto">if [[ github.event.inputs.env_name == "dev" ]]; then 
</code></pre>
<p>(and the same in the <code>elif</code>). You’re comparing the literal string “github.event.inputs.env_name”, not the value of the input, so neither case runs. If you had added an else clause like</p>
<pre><code class="lang-auto">else
  echo "::error::Unrecognized env_name"
  exit 1
</code></pre>
<p>it would have been a little more obvious where it was going wrong.</p>
<p>To substitute the value of the input into the script, you need to do like</p>
<pre><code class="lang-auto">if [[ ${{ github.event.inputs.env_name }} == "dev" ]]; then 
</code></pre>
<p>Although personally I’m always wary of substituting text directly into the bash script, as it seems too easy to accidentally break if the value is ever something unexpected. So I’d prefer to do like</p>
<pre><code class="lang-auto">env:
  ENV_NAME: ${{ github.event.inputs.env_name }}
run: |
  if [[ $ENV_NAME == "dev" ]]; then
    ...
</code></pre>
